x/simulation/account.go

```diff
@@ -45,3 +46,45 @@ func RandomAccounts(r *rand.Rand, n int) []Account {
 
 	return accs
 }
+
+// FindAccount iterates over all the simulation accounts to find the one that matches
+// the given address
+func FindAccount(accs []Account, address sdk.AccAddress) (Account, bool) {
+	for _, acc := range accs {
+		if acc.Address.Equals(address) {
+			return acc, true
+		}
+	}
+
+	return Account{}, false
+}
+
+// RandomFees returns a random fee by selecting a random coin denomination and
+// amount from the account's available balance. If the user doesn't have enough
+// funds for paying fees, it returns empty coins.
+func RandomFees(r *rand.Rand, ctx sdk.Context, spendableCoins sdk.Coins) (sdk.Coins, error) {
+	if spendableCoins.Empty() {
+		return nil, nil
+	}
+
+	denomIndex := r.Intn(len(spendableCoins))
+	randCoin := spendableCoins[denomIndex]
+
+	if randCoin.Amount.IsZero() {
+		return nil, nil
+	}
+
+	amt, err := RandPositiveInt(r, randCoin.Amount)
+	if err != nil {
+		return nil, err
+	}
+
+	// Create a random fee and verify the fees are within the account's spendable
+	// balance.
+	fees := sdk.NewCoins(sdk.NewCoin(randCoin.Denom, amt))
```

Comment:
How would this case ever evaluate? If we know spendable coins is non-zero and we pick a random amount (X > 0) from spendable coins (Y), how can Y - X < 0, were 0 < X <= Y?